wire-library/wire-runtime-swift/src/main/swift/ProtoDecoder.swift

```diff
@@ -96,6 +97,8 @@ public final class ProtoDecoder {
                 return "Unknown case with value \(fieldNumber) found for enum of type \(String(describing: type))."
             case let .unterminatedGroup(fieldNumber):
                 return "The group with field number \(fieldNumber) has no matching end-group key."
+            case .unexpectedCallToBeginMessage:
+                return "Unexpected call to beginMessage()."
```

Comment:
Can we be more descriptive of what's wrong with the message data here (and in the error name). Since this is unlikely to be called from anything but Wire-generated code, it's more likely that the data is bad and not that someone accidentally called `beginMessage` unexpectedly.

In this case I think it's something like:

```
case .messageWithoutLength:
  return "Attempting to decode a message without first decoding the length of that message."
```

I'd say we should include the field number, but if we don't have the length then we almost certainly don't have the field number either, so this seems as good as we can do until we better track the actual byte offset.

Reply:
Yeah, great point. That message sounds good to me -- I'll update.